frontend/packages/dev-console/src/components/topology/components/MoveConnectionModal.tsx

```diff
@@ -0,0 +1,171 @@
+      <ModalBody>
+        <Title size="sm" className="co-m-form-row">
+          Connect
+          <strong>{` ${edge.getSource().getLabel()} `}</strong>
+          to
+        </Title>
+        <FormSection fullWidth>
+          <FormGroup fieldId="target-node" label="Target">
+            <Dropdown
+              id="target-node-dropdown"
+              className="dropdown--full-width"
+              toggle={
+                <DropdownToggle id="toggle-id" onToggle={onToggle} iconComponent={CaretDownIcon}>
+                  {nodeItem(values.target)}
+                </DropdownToggle>
+              }
+              isOpen={isOpen}
+              dropdownItems={availableTargets.map(dropDownNodeItem)}
+            />
+          </FormGroup>
+        </FormSection>
+      </ModalBody>
+      <ModalSubmitFooter
+        submitText="Move"
+        submitDisabled={!isDirty}
+        cancel={cancel}
+        inProgress={isSubmitting}
+        errorMessage={status && status.submitError}
+      />
+    </form>
+  );
+};
+
+class MoveConnectionModal extends PromiseComponent<
+  MoveConnectionModalProps & StateProps,
+  MoveConnectionModalState
+> {
+  private onSubmit = (newTarget: Node): Promise<K8sResourceKind[] | K8sResourceKind> => {
+    const { edge, serviceBinding } = this.props;
+    switch (edge.getType()) {
+      case TYPE_CONNECTS_TO:
+        return createConnection(edge.getSource(), newTarget, edge.getTarget(), serviceBinding);
+      case TYPE_SERVICE_BINDING:
+        return createConnection(edge.getSource(), newTarget, edge.getTarget(), serviceBinding);
+      case TYPE_EVENT_SOURCE_LINK:
+        return createSinkConnection(edge.getSource(), newTarget);
+      default:
+        return Promise.reject(new Error(`Unable to move connector of type ${edge.getType()}.`));
+    }
+  };
+
+  private handleSubmit = (values, actions) => {
+    actions.setSubmitting(true);
+    const { close } = this.props;
+    this.handlePromise(this.onSubmit(values.target))
+      .then(() => {
+        actions.setSubmitting(false);
+        close();
+      })
+      .catch((err) => {
+        actions.setSubmitting(false);
+        actions.setStatus({ submitError: err });
+      });
+  };
+
+  render() {
+    const { edge } = this.props;
+    const initialValues = {
+      target: edge.getTarget(),
+    };
+    return (
+      <Formik
+        initialValues={initialValues}
+        onSubmit={this.handleSubmit}
+        render={(formProps) => <MoveConnectionForm {...formProps} {...this.props} />}
+      />
+    );
+  }
+}
+
+const getServiceBindingStatus = ({ FLAGS }: RootState): boolean => FLAGS.get(ALLOW_SERVICE_BINDING);
+
+const mapStateToProps = (state: RootState): StateProps => {
+  return {
+    serviceBinding: getServiceBindingStatus(state),
+  };
+};
+
+const ConnectedMoveConnectionModal = connect(mapStateToProps)(MoveConnectionModal);
+
+export const moveConnectionModal = createModalLauncher((props: MoveConnectionModalProps) => (
```

Comment:
😕 There are a lot of different ways to make modals in our codebase. I'll need to take an action up to talk to Christian and find out what the best way is and try to centralize around it.